lib/Conversion/TorchOnnxToTorch/DefaultDomainGtoP.cpp

```diff
@@ -335,38 +335,66 @@ void mlir::torch::onnx_c::populateDefaultDomainGtoP(
                       binder.op, resultType, lhs, rhs);
                   return success();
                 });
-  patterns.onOp(
-      "Max", 1, [](OpBinder binder, ConversionPatternRewriter &rewriter) {
-        Torch::ValueTensorType resultType;
-        llvm::SmallVector<Value> operands;
-        if (binder.tensorOperandsList(operands) ||
-            binder.tensorResultType(resultType) || operands.size() == 0) {
-          return failure();
-        }
-        Value result = operands[0];
-        for (uint64_t i = 1; i < operands.size(); i++) {
-          result = rewriter.create<Torch::AtenMaximumOp>(
-              binder.getLoc(), resultType, result, operands[i]);
-        }
-        rewriter.replaceOp(binder.op, result.getDefiningOp());
-        return success();
-      });
-  patterns.onOp(
-      "Min", 1, [](OpBinder binder, ConversionPatternRewriter &rewriter) {
-        Torch::ValueTensorType resultType;
-        llvm::SmallVector<Value> operands;
-        if (binder.tensorOperandsList(operands) ||
-            binder.tensorResultType(resultType) || operands.size() == 0) {
-          return failure();
-        }
-        Value result = operands[0];
-        for (uint64_t i = 1; i < operands.size(); i++) {
-          result = rewriter.create<Torch::AtenMinimumOp>(
-              binder.getLoc(), resultType, result, operands[i]);
-        }
-        rewriter.replaceOp(binder.op, result.getDefiningOp());
-        return success();
-      });
+  patterns.onOp("InstanceNormalization", 6,
+		 [](OpBinder binder, ConversionPatternRewriter &rewriter) {
+		   Torch::ValueTensorType resultType;
+                   llvm::SmallVector<Value> operands;
+		   float eps;
+
+		   if (binder.tensorOperands(operands, 3) ||
+                       binder.tensorResultType(resultType) ||
+                       operands.size() != 3 || 
+		       binder.f32FloatAttr(eps, "epsilon", 1e-05f)) {
+                     return failure();
+                   }
+
+		   auto epsValue = rewriter.create<Torch::ConstantFloatOp>(binder.getLoc(),
+                                     rewriter.getF64FloatAttr(eps));
+                   
+                   auto outputScale = rewriter.create<Torch::ConstantFloatOp>(binder.getLoc(),
+                                        rewriter.getF64FloatAttr(1.0f));
+                   auto outputZeroPoint = rewriter.create<Torch::ConstantIntOp>(
+                                            binder.getLoc(), rewriter.getI64IntegerAttr(0));
+	           rewriter.replaceOpWithNewOp<Torch::QuantizedInstanceNormOp>(
```

Comment:
Is this the right op to decompose into? From looking at the [ONNX docs for `InstanceNormalization`](https://onnx.ai/onnx/operators/onnx__InstanceNormalization.html), it seems that it performs the same computation as the `torch.nn.functional.batch_norm` op, and we already have support for `batch_norm` in torch-mlir

Reply:
[the paper ](https://arxiv.org/pdf/1701.02096.pdf) says that batch normalization "applies the normalization to a whole batch of images instead of single ones."

Reply:
I'm not familiar with the paper. From the ONNX docs, `InstanceNormalization` takes inputs of shape `(NxCx...)`, where `N` is the batch dimension, so `torch.nn.functional.batch_norm` should be a good match.

Reply:
from what I understand, instance norm calculates the mean and variance per image, and then normalizes with that instead of where batch normalization does by batches.  The onnx op carries instance normalization based on that paper, but they are different ops.

Reply:
Ah, I see what you mean. You are right, they do different things. PyTorch does have an instance version as well: `'aten::instance_norm`. We should be using this one rather than the quantized one. This op should be testable e2e.

Reply:
aten::instance_norm op uses running mean, running var, and momentum unlike the onnx.InstanceNormalization. quantized::instance_norm has zero point and output_scale, but the rest of the args are the same as the onnx instance normalization.

Reply:
> aten::instance_norm op uses running mean, running var, and momentum

That's fine, the running statistics are optional. Your lowering can just match when they are `None`. The `momentum` can be ignored by being set to 0, I believe.

My opposition to using the quantized one is that `onnx.InstanceNormalization` is not a quantized operation. From the docs, the ONNX op takes regular float tensors and produces regular float tensors. On the other hand, the PyTorch version (basing from [the quantized::batch_norm docs](https://pytorch.org/docs/stable/generated/torch.quantized_batch_norm.html), since there are no docs for quantized::instance_norm) takes quantized tensors as inputs (tensors with dtype `torch.q...`). So, it is not the right operation to lower `onnx.InstanceNormalization` to.

Reply:
Ok. the e2e test fails because it says mulf expects a floating point like operand but gets a torch.none, but why is the instance norm e2e test failing?

Reply:
The e2e instance norm test is failing because the conversion of `instance_norm` to linalg is generating invalid IR. In particular, it is generating a `mulf` op with invalid inputs.